pkg/network/protocols/http/shared_libraries.go

```diff
@@ -95,75 +94,65 @@ type soRule struct {
 // soWatcher provides a way to tie callback functions to the lifecycle of shared libraries
 type soWatcher struct {
 	procRoot       string
-	all            *regexp.Regexp
 	rules          []soRule
 	loadEvents     *ddebpf.PerfHandler
 	processMonitor *monitor.ProcessMonitor
 	registry       *soRegistry
 }
 
+type pathIdentifierSet = map[pathIdentifier]struct{}
+
 type soRegistry struct {
 	m     sync.Mutex
 	byID  map[pathIdentifier]*soRegistration
-	byPID map[uint32]*soRegistration
+	byPID map[uint32]pathIdentifierSet
 
 	// if we can't register a uprobe we don't try more than once
-	blocklistByID map[pathIdentifier]struct{}
+	blocklistByID pathIdentifierSet
 }
 
 func newSOWatcher(perfHandler *ddebpf.PerfHandler, rules ...soRule) *soWatcher {
-	allFilters := make([]string, len(rules))
-	for i, r := range rules {
-		allFilters[i] = r.re.String()
-	}
-
-	all := regexp.MustCompile(fmt.Sprintf("(%s)", strings.Join(allFilters, "|")))
 	return &soWatcher{
 		procRoot:       util.GetProcRoot(),
-		all:            all,
 		rules:          rules,
 		loadEvents:     perfHandler,
 		processMonitor: monitor.GetProcessMonitor(),
 		registry: &soRegistry{
 			byID:          make(map[pathIdentifier]*soRegistration),
-			byPID:         make(map[uint32]*soRegistration),
-			blocklistByID: make(map[pathIdentifier]struct{}),
+			byPID:         make(map[uint32]pathIdentifierSet),
+			blocklistByID: make(pathIdentifierSet),
 		},
 	}
 }
 
 type soRegistration struct {
-	pathID       pathIdentifier
-	refcount     int
-	unregisterCB func(pathIdentifier) error
+	uniqueProcessesCount int
+	unregisterCB         func(pathIdentifier) error
 }
 
-// Unregister return true if there are no more reference to this registration
-func (r *soRegistration) Unregister() bool {
-	r.refcount--
-	if r.refcount > 0 {
+// unregister return true if there are no more reference to this registration
+func (r *soRegistration) unregister(pathID pathIdentifier) bool {
+	r.uniqueProcessesCount--
+	if r.uniqueProcessesCount > 0 {
 		return false
 	}
 	if r.unregisterCB != nil {
-		if err := r.unregisterCB(r.pathID); err != nil {
-			log.Debugf("error while unregistering %s : %s", r.pathID.String(), err)
+		if err := r.unregisterCB(pathID); err != nil {
+			// Even if we fail here, we have to return true, as best effort methodology.
+			// We cannot handle the failure, and thus we should continue.
+			log.Debugf("error while unregistering %s : %s", pathID.String(), err)
```

Comment:
Would it make sense to log it at warn level? An error returned from `unregisterCB` is likely to indicate some system-level issue an operator would want to be aware of